app/pastes/pasteRepository.js

```diff
@@ -1,11 +1,32 @@
 class PasteRepository {
-    constructor({ dbProvider, pasteFactory }) {
+    constructor({ dbProvider, pasteFactory, cacheProvider }) {
         this.dbProvider = dbProvider;
         this.pasteFactory = pasteFactory;
+        this.cacheProvider = cacheProvider;
     }
 
     async findById({ id }) {
+        const isPasteCached = await this.cacheProvider.exists(`cached_${id}`);
+        if (isPasteCached) {
+            const cachedData = JSON.parse(await this.cacheProvider.get(`cached_${id}`));
+            //console.log('cache is used');
+            return this.pasteFactory.create({
```

Comment:
Dont u have a dedicated method for this?

Reply:
dedicated method is used for findAll where i get data from db and it uses db column names like 'total_views', here i use '_totalViews'. I can implement it but i the dedicated methos will be suitable either for arrays if we use it it findAll method and uses db fields or for non arrays and uses '_something' fields